plugins/main/public/components/common/doc_viewer/doc_viewer.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { escapeRegExp } from 'lodash';
+import { i18n } from '@osd/i18n';
+import { FieldIcon } from '../../../../../../src/plugins/opensearch_dashboards_react/public';
+import { EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
+
+const COLLAPSE_LINE_LENGTH = 350;
+const DOT_PREFIX_RE = /(.).+?\./g;
+
+export type tDocViewerProps = {
+    flattened: any;
+    formatted: any;
+    mapping: any;
+    indexPattern: any;
+}
+
+/**
+ * Convert a dot.notated.string into a short
+ * version (d.n.string)
+ */
+export const shortenDottedString = (input: string) => input.replace(DOT_PREFIX_RE, '$1.');
+
+export function getFieldTypeName(type: string) {
```

Comment:
If "const" is used above for functions, then I suggest using the same for all functions

Reply:
Yes, that's right. I'd replaced the function by const